packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx

```diff
@@ -210,18 +206,26 @@ const MetadataLabeling = (props: Props) => {
     const metadata = useSelector(state => state.metadata);
     const { isDiscoveryRunning, device } = useDiscovery();
     const [showSuccess, setShowSuccess] = useState(false);
+    const [pending, setPending] = useState(false);
```

Comment:
merge with `isSubscribedToSubmitResult`, use string or undefined

Reply:
I am starting to think it is not possible to merge these two variables `pending` which is a react component state and `isSubscribedToSubmitResult` which is a ref into one. Because:
1. if I use only state, it won't be accessible in event handler right after it was updated because the handler function has closure over current value of pending only. 
1. if I use only ref and turn it from bool to string I will run into another problem - changing ref does not trigger rerender of component.